### PR TITLE
Added sonarcloud integration to readme

### LINKED ISSUE #29: Document SonarCloud integration
It might be worth documenting how SonarCloud (and SonarCube may be) can be configured so see these rules.

### PR DESCRIPTION
**Pull request by Gemberkoekje**
Closes #29

### CODE REVIEW

**Comment by Corniel** [comment]
I think is deserves its own page. Location could be `general\sonar-integration.md`?! May be also shortly explain what the problem is, and some links to Sonar?!

**Comment by Gemberkoekje** [comment]
> I think is deserves its own page. Location could be `general\sonar-integration.md`?! May be also shortly explain what the problem is, and some links to Sonar?!

Updated.

File: README.md [hunk 1]
```diff
@@ -143,13 +143,3 @@ To add a props file:
 </Project>
 ```
```
Comment (Corniel) [inline README.md]:
```suggestion

## Sonar integration
By default, results by .NET project file analyzers are not added to Sonar's reporting. Read [here](general/sonar-integration.md) how to configure this correctly.
```

File: general/sonar-integration.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Sonarcloud
+
+*Currently, this readme is confirmed to work for SonarCloud. The steps in this guide
+might also be relevant for other [SonarSource](https://www.sonarsource.com) products,
+such as SonarQube, but this has not been verified at this time*
+
+[Sonarcloud](https://www.sonarsource.com/products/sonarcloud/) is a service which allows
+you to scan your code for issues, and gives an online dashboard allowing you to see which
+issues exist within the code. This includes any warnings generated by analyzers. In addition,
+it has integration with DevOps platforms, which can be set up to automatically add comments
```
Comment (Corniel) [inline general/sonar-integration.md:10]: (Such as GitHub and Azure)

File: general/sonar-integration.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+on pull requests on any new warnings that the change has made, which can help in improving
+code quality.
+
+## The issue with this analyzer and Sonarcloud
+
+Sonarcloud's analyzer software can run while a build is being made, for example as part
+of a Pull Request check. However, Sonarcloud, by default, does not pick up any warnings
+generated by ??proj files. This means that any warnings generated by this analyzer will
```
Comment (Corniel) [inline general/sonar-integration.md:18]:
```suggestion
generated by `.??proj` and `.props` files. This means that any warnings generated by this analyzer will
```

File: general/sonar-integration.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+not generate comments on pull requests, which could result in a situation where they are
```
Comment (Corniel) [inline general/sonar-integration.md:19]:
```suggestion
not generate comments on pull requests, nor with they show up in their reports, which could result in a situation where they are
```

File: general/sonar-integration.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+not noticed and solved by developers.
+
+## Sonarcloud integration
+There is currently no 'elegant' solution to have the warnings show up on SonarCloud,
```
Comment (Corniel) [inline general/sonar-integration.md:23]:
```suggestion
There is currently no *'elegant'* solution to have the warnings show up on SonarCloud,
```

File: general/sonar-integration.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+however, there is a workaround. By adding the ??proj file as content, SonarCloud will
+recognise it's warnings and pick them up, same as any warnings in the code.
+
+For Example:
+``` XML
+  <ItemGroup>
+    <Content Include="$(MSBuildProjectFile)" CopyToOutputDirectory="Never"/>
```
Comment (Corniel) [inline general/sonar-integration.md:30]: This wil only add the project file, not the props files.

Beter might be: 
``` XML
  <Content Include="{path to props}" CopyToOutputDirectory="Never"/>
  <Content Include="*.??proj" CopyToOutputDirectory="Never"/>
 ```
 
 But I'm not sure how exactly yet.

**Review by Corniel — CHANGES_REQUESTED** [review]
Start looking promising. But not ready yet.

File: general/sonar-integration.md (hunk 6 above)
Comment (Gemberkoekje) [reply general/sonar-integration.md:30]: I'm not sure adding the props file separately is necessary. I'll do a test tomorrow to see whether just adding the csproj files also picks up the props files (as the warnings are generated in the analysis of the csproj files, not the props files themselves)

File: README.md (hunk 1 above)
Comment (Gemberkoekje) [reply README.md:145]: Done.

File: general/sonar-integration.md (hunk 2 above)
Comment (Gemberkoekje) [reply general/sonar-integration.md]: Done.

File: general/sonar-integration.md (hunk 3 above)
Comment (Gemberkoekje) [reply general/sonar-integration.md]: Done.

File: general/sonar-integration.md (hunk 4 above)
Comment (Gemberkoekje) [reply general/sonar-integration.md]: Done.

File: general/sonar-integration.md (hunk 5 above)
Comment (Gemberkoekje) [reply general/sonar-integration.md]: Done.

File: general/sonar-integration.md (hunk 6 above)
Comment (Corniel) [reply general/sonar-integration.md:32]: I guess not. Only the files in the content are included from what I understood.

Comment (Gemberkoekje) [reply general/sonar-integration.md:33]: You were correct, appended the readme to reflect that.

File: general/sonar-integration.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+## Sonarcloud integration
+There is currently no *'elegant'* solution to have the warnings show up on SonarCloud,
+however, there is a workaround. By adding the `.??proj` and `.props` files as content,
+SonarCloud will recognise it's warnings and pick them up, same as any warnings in the
+code.
+
+For Example:
+``` XML
+  <ItemGroup>
+    <Content Include="$(MSBuildProjectFile)" CopyToOutputDirectory="Never"/>
```
Comment (Corniel) [inline general/sonar-integration.md:33]: I prefer:
```suggestion
    <Content Include="*.??proj" CopyToOutputDirectory="Never"/>
```

The reason is that this also they way it is advised to include project files as `<AdditionalFiles>` in the rest of the documentation.

File: general/sonar-integration.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# Sonarcloud
+
```
Comment (Corniel) [inline general/sonar-integration.md:2]: please remove the empty lines between headers and following paragraph.

**Review by Corniel — APPROVED** [review]
Some minor left-overs. But after that, ready to be merged.

File: general/sonar-integration.md (hunk 7 above)
Comment (Gemberkoekje) [reply general/sonar-integration.md]: Updated

File: general/sonar-integration.md (hunk 8 above)
Comment (Gemberkoekje) [reply general/sonar-integration.md]: Done

**Pull request merged**